fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/EntityTeleportationHelper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.entity.event.v1;
+
+import java.util.Optional;
+
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.mob.PathAwareEntity;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.MathHelper;
+
+/**
+ * A utility class providing methods to help teleport entities to other server worlds.
+ */
+public final class EntityTeleportationHelper {
```

Comment:
Hmm, what's the point of Area Helper from minecraft then

Reply:
AreaHelper does not teleport the entity. The only teleport logic is Server player, entity moveToWorld and teleport command.

However an event for the after world change is gonna be a pain for other mod devs since they would have to go pretty manual. This just makes a lazy one method call to teleport an entity straight to another world.

Reply:
I feel like this class should be in another module as it's more generalized than just for entity events.

Reply:
I believe that a method in updated dimension api pull request would suffice here and reduces duplicate logic.